crates/ra_hir_def/src/nameres/collector.rs

```diff
@@ -854,6 +878,28 @@ impl ModCollector<'_, '_> {
         }
     }
 
+    fn collect_proc_macro(&mut self, attrs: &Attrs) {
+        if let Some(derive_subtree) = attrs.by_key("proc_macro_derive").tt_values().next() {
```

Comment:
Why do we need to collect proc_macro *definitions* here? 

I think the more correct approach would be to collect the set of procmacros from the compiled proc_macro code.

For example, if we had `foo` proc macro, than run `cargo check` to compile it, and then rename it to `bar`, *without* re-runing `cargo check`, we should resole and epxand `foo` and not `bar`.

Reply:
Um.. the original idea is, we might want to get back some location information from the proc-macro definition in the feature, but I think your approach is reasonable too.

Reply:
Yeah, I think long-term we need to do both (and, in general, have capabilities for relating compiled code (proc macros and/or rlibs) and source code), but it's important to not mix the two accidently, so I'd skip this for now.